Update rails to 7.1.3.4 and replace dotenv-rails with dotenv

dmitryrck opened the pull request:
Main things:

* Update rails
* Replace dotenv-rails with dotenv (See https://github.com/bkeepers/dotenv/pull/468 )
* Update all of the other gems
* Fix some controllers tests and add a missing one

Honestly, I am not too happy with the way I fixed the controllers tests. Rails probably changed and it needs now the ENV var at boot. Happy to have someone else trying.

:v:

sarahhay7 reviewed (approved):
Thanks for going through this @dmitryrck  ✨ 

It all looks good to me. We have previously used the :destroy action on the members controller in a script for culling members who have been inactive for 3+ years, but I think it's right to remove it from the controller. We can add back it in future if needed (or just modify the database directly).

sarahhay7 commented:
@dmitryrck I only noticed one controller test being added. Is it possible there are more commits that haven't been included in this PR?

dmitryrck commented:
> Thanks for going through this @dmitryrck ✨

🙏 

> […] We have previously used the :destroy action on the members controller in a script for culling members who have been inactive for 3+ years.

If I recall that correctly it was done using `rails console`, there is no action `destroy` in that controller. I removed it from that list because rails started to complain that I don't have an action `destroy`.

> @dmitryrck I only noticed one controller test being added. Is it possible there are more commits that haven't been included in this PR?

Nop. The tests already in the `main` branch only test the "best case scenario". I added a "wrong case scenario" to make sure the authentication is blocking users with wrong password.

Thanks for the review :)

sarahhay7 commented:
@dmitryrck You're totally right 😅 Can't believe that was only 3 years ago 👀 
And thanks again 🙌

parndt commented:
I'll get this deployed
